nhrpd/nhrp_cache.c

```diff
@@ -119,12 +119,50 @@ static void nhrp_cache_update_route(struct nhrp_cache *c)
 {
 	struct prefix pfx;
 	struct nhrp_peer *p = c->cur.peer;
+	char buf[3][SU_ADDRSTRLEN];
+	struct nhrp_interface *nifp;
 
 	sockunion2hostprefix(&c->remote_addr, &pfx);
 
 	if (p && nhrp_peer_check(p, 1)) {
-		netlink_update_binding(p->ifp, &c->remote_addr,
-				       &p->vc->remote.nbma);
+		if (sockunion_family(&c->cur.remote_nbma_natoa) != AF_UNSPEC) {
+			/* remote nbma is already set. Therefore, this is a
+			 * shortcut entry. Peer should NOT be changed to the
+			 * vc's remote nbma.
+			 */
```

Comment:
not sure the comment is related to the modification. you mean that if there is a nat device in between ,you update the link address. right? if yes, then shortcut term should not be used, there.

Reply:
right, updating the comments.